django/contrib/admin/widgets.py

```diff
@@ -194,10 +195,29 @@ def label_for_value(self, value):
         key = self.rel.get_related_field().name
         try:
             obj = self.rel.model._default_manager.using(self.db).get(**{key: value})
-            return '&nbsp;<strong>%s</strong>' % escape(Truncator(obj).words(14, truncate='...'))
-        except (ValueError, self.rel.model.DoesNotExist):
+        except self.rel.model.DoesNotExist:
             return ''
 
+        label = "&nbsp;<strong>{}</strong>"
+        text = Truncator(obj).words(14, truncate='...')
+        try:
+            change_url = reverse(
+                '%s:%s_%s_change' % (
+                    self.admin_site.name,
+                    obj._meta.app_label,
+                    obj._meta.object_name.lower(),
+                ),
+                args=(obj.pk,)
+            )
+        except NoReverseMatch:
+            # Admin not registered for target model:
+            return format_html(label, text)
+        else:
```

Comment:
You could put `text = format_html('<a href="{}">{}</a>', change_url, text)` in the "else" block, `pass` in the `except` block and then a single return statement at the end `return format_html(label, text)`.